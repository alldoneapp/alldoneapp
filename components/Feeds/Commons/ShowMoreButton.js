import React from 'react'
import { StyleSheet } from 'react-native'
import { TouchableOpacity } from 'react-native-gesture-handler'

import { colors } from '../../styles/global'
import Icon from '../../Icon'

export default function ShowMoreButton({ forExpand, onPress, style }) {
    return (
        <TouchableOpacity style={[localStyles.button, { marginBottom: forExpand ? 16 : 0 }, style]} onPress={onPress}>
            <Icon name={forExpand ? 'chevron-down' : 'chevron-up'} size={24} color={colors.Text04} />
        </TouchableOpacity>
    )
}

const localStyles = StyleSheet.create({
    button: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        alignSelf: 'center',
        padding: 8,
        height: 40,
    },
})
