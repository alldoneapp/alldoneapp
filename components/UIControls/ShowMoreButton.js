import React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'

import styles, { colors } from '../styles/global'
import Icon from '../Icon'
import { translate } from '../../i18n/TranslationService'

export default function ShowMoreButton({ expanded, contract, expand, style, expandText, contractText }) {
    return (
        <View style={[localStyles.moreButton, style]}>
            <TouchableOpacity style={localStyles.container} onPress={expanded ? contract : expand}>
                {expandText || contractText ? (
                    <Text style={localStyles.text}>{translate(expanded ? contractText : expandText)}</Text>
                ) : (
                    <Icon name={expanded ? 'chevron-up' : 'chevron-down'} size={24} color={colors.Text04} />
                )}
            </TouchableOpacity>
        </View>
    )
}

const localStyles = StyleSheet.create({
    container: {
        justifyContent: 'center',
    },
    moreButton: {
        flex: 1,
        height: 24,
        flexDirection: 'row',
        justifyContent: 'center',
    },
    text: {
        ...styles.buttonLabel,
        fontFamily: 'Roboto-Regular',
        color: colors.Text03,
    },
})
