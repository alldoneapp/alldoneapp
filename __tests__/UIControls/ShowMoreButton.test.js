/**
 * @jest-environment jsdom
 */

import React from 'react'
import ShowMoreButton from '../../components/UIControls/ShowMoreButton'

import renderer from 'react-test-renderer'

describe('Show More Button component', () => {
    describe('Task Estimation Button snapshot test', () => {
        it('should render correctly', () => {
            const tree = renderer
                .create(<ShowMoreButton expanded={false} amount={1} expand={() => {}} contract={() => {}} />)
                .toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
