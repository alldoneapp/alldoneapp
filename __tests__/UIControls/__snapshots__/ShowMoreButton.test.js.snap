// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show More Button component Task Points Button snapshot test should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "marginBottom": 0,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "padding": 8,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#B7BDC8",
            "fontFamily": "alldone",
            "fontSize": 24,
            "opacity": 1,
          }
        }
      >
        î¤°
      </Text>
    </View>
  </View>
</View>
`;
